### PR TITLE
Copying in SqlDelight's Error Handler into Skate

### PR DESCRIPTION
**Pull request by kateliu20**
First copying in and wiring up [SqlDelight's error handler](https://github.com/cashapp/sqldelight/blob/5512326251b1e9f91ddef49dda75d27405943e2f/sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/SqlDelightErrorHandler.kt#L35). 

and using placeholders for the `GIT_SHA`, `VERSION`, and `BUGSNAG_KEY`

### CODE REVIEW

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent
+import com.intellij.openapi.diagnostic.SubmittedReportInfo
+import com.intellij.util.Consumer
+import java.awt.Component
+
+/**
+ * Using SqlDelightErrorHandler to use for crash reporting for the Skate plugin. Currently just have
+ * placeholders for GIT_SHA, VERSION, and BUGSNAG_KEY
+ */
+class SqlDelightErrorHandler : ErrorReportSubmitter() {
```
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt:33]: Please update the names for everything to align with the skate plugin, not sqldelight

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016 Slack Technologies, LLC
```
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt:2]: We'll need to preserve the original copyright header

File: skate-plugin/src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -30,8 +30,10 @@
                 serviceImplementation="com.slack.sgp.intellij.SkateProjectServiceImpl"/>
         <projectService serviceImplementation="com.slack.sgp.intellij.SkatePluginSettings"/>
         <projectConfigurable instance="com.slack.sgp.intellij.SkateConfig"/>
+        <errorHandler implementation="com.slack.sgp.intellij.SqlDelightErrorHandler"/>
     </extensions>
 
+
```
Comment (ZacSweers) [inline skate-plugin/src/main/resources/META-INF/plugin.xml:36]: ?

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+class SqlDelightErrorHandler : ErrorReportSubmitter() {
+  val skateBugsnagApiKey = "BUGSNAG_KEY_PLACEHOLDER"
+
+  val skateBugsnag = Bugsnag(skateBugsnagApiKey, false)
+
+  init {
+    skateBugsnag.setAutoCaptureSessions(false)
+    skateBugsnag.startSession()
+    skateBugsnag.setAppVersion("VERSION_PLACEHOLDER")
+    skateBugsnag.setProjectPackages("app.cash.sqldelight")
```
Comment (ZacSweers) [inline skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt:42]: Update this to our package name

Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt]: Got it: 267ae320006591e5039c6335689321f06e0fbb6b

File: skate-plugin/src/main/resources/META-INF/plugin.xml (hunk 3 above)
Comment (kateliu20) [reply skate-plugin/src/main/resources/META-INF/plugin.xml]: Think I added an extra line 😅  c27e74dd7e7c530520f1db46a1ed1e4913cd0a10

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt (hunk 2 above)
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SqlDelightErrorHandler.kt]: 36c089b8d35bdf57ba506ec09db5c411b331fa27

**Review by ZacSweers — APPROVED** [review]

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateErrorHandler.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2016 Slack Technologies, LLC
```
[the remark this answers is not in the document]
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateErrorHandler.kt:2]: Should I reinstate the original header or have that be in a followup?

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateErrorHandler.kt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.diagnostic.ErrorReportSubmitter
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent
+import com.intellij.openapi.diagnostic.SubmittedReportInfo
+import com.intellij.util.Consumer
+import java.awt.Component
+
+/**
+ * Using SqlDelightErrorHandler to use for crash reporting for the Skate plugin. Currently just have
```
[the remark this answers is not in the document]
Comment (kateliu20) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateErrorHandler.kt:30]: 512300be1eef9fc570ecd33ec1a5ca7b2bd957ad

File: skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateErrorHandler.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (ZacSweers) [reply skate-plugin/src/main/kotlin/com/slack/sgp/intellij/SkateErrorHandler.kt:2]: I'll handle it 👍

**Pull request merged**